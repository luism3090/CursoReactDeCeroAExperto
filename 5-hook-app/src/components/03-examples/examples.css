body{
    padding: 70px;
    color: red;
}