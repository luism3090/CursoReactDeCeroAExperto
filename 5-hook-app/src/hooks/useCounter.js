import { useState } from "react"

export const useCounter = (initialState = 10) => {


    const [state, setState] = useState(initialState);

    const incrementar = () =>{
        setState(state + 1);
    }

    const decrementar = () =>{
        setState(state - 1);
    }

    const incrementarWithFactor = (factor) =>{
        setState(state + factor);
    }

    const decrementarWithFactor = (factor) =>{
        setState(state - factor);
    }

    const reset = () =>{
        setState(initialState);
    }


    return {
        state,
        incrementar,
        decrementar,
        incrementarWithFactor,
        decrementarWithFactor,
        reset
    }

}
