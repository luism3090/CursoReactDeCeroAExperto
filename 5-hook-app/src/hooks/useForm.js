import React, { useState } from 'react'

export const useForm = (initialState = {}) => {
    
    //console.log("entro al useForm");

    const [formStateValues, setFormStateValues] = useState(initialState);
    
    const reset = () => {
        setFormStateValues(initialState);
    }

    const changeValueInput = ({target}) => {
        
        setFormStateValues({
            ...formStateValues,
            [ target.name ]: target.value
        });
    }

    return [formStateValues, changeValueInput, reset];

}
