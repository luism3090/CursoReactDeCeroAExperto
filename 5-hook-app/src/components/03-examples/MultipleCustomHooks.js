import React from 'react'
import { useCounter } from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch';
import './examples.css';

export const MultipleCustomHooks = () => {

    const {state:counter, incrementar} = useCounter(1);

    const {data, loading} = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);
    
    const {author, quote} = (data !== null) && data[0];

    // console.log(author);
    // console.log(quote);
    // console.log(quote_id);

    return (


        <div>
            <h1>Breaking Bad</h1>
            <hr/>
        {
            loading 
            ?
                (
                    <div className='alert'> loading... </div>
                )
            :
            (
                <blockquote>
                <p>{quote}</p>
            <footer>{author}</footer>
            </blockquote>
            )
        }
            
        <button className="btn btn-primary" onClick={incrementar}>
            Siguiente quote
        </button>
        

        </div>

    )
}
