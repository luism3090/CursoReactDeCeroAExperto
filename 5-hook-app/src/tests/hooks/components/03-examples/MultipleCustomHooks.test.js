import React from 'react'
import {shallow} from 'enzyme';
import { MultipleCustomHooks } from '../../../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../../../hooks/useFetch';
jest.mock('../../../../hooks/useFetch');

describe('preubas en MultipleCustomHooks', () => {
    
    test('Debe de tomarse una foto snapshot', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });
        
        const wrapper = shallow(<MultipleCustomHooks />);

        expect(wrapper).toMatchSnapshot();

    })

    test('Debe de mostrar la informacion', () => {
        
        useFetch.mockReturnValue({
            data: [{
                author: 'Fernando',
                quote: 'Hola mundo'
            }],
            loading: false,
            error: null
        });


        const wrapper = shallow(<MultipleCustomHooks />);

        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('p').text().trim()).toBe( 'Hola mundo' );
        expect(wrapper.find('footer').text().trim()).toBe( 'Fernando' );

        //console.log(wrapper.find('p').text().trim());

    })
    

});
