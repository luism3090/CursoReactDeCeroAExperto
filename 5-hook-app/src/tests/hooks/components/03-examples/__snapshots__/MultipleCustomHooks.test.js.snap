// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preubas en MultipleCustomHooks Debe de tomarse una foto snapshot 1`] = `
<div>
  <h1>
    Breaking Bad
  </h1>
  <hr />
  <div
    className="alert"
  >
     loading... 
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</div>
`;
